lib/host/wasi/wasifunc.cpp

```diff
@@ -560,6 +560,13 @@ Expect<uint32_t> WasiFdAllocate::body(const Runtime::CallingFrame &, int32_t Fd,
 Expect<uint32_t> WasiFdClose::body(const Runtime::CallingFrame &, int32_t Fd) {
   const __wasi_fd_t WasiFd = Fd;
 
+  if (auto Poll = this->Env.acquirePollerOnly(); unlikely(!Poll)) {
```

Comment:
This pattern is weird, maybe refactoring this one with the above comments can help.

Reply:
I use single close function instead.